Ark esc docs (#26132)
PX4/PX4-Autopilot: docs/en/esc/esc_protocols.md

> # ESC Protocols

Reviewer: FYI all same content, but have reorded to have Dshot and DroneCAN first - somewhat cooincidental, since I wnted them first and alphabetic ordering makes that happen. We should add Cyphal here

This topic lists the main [Electronic Speed Controller (ESC)](../peripherals/esc_motors.md) protocols supported by PX4.

## DShot

> [DShot](../peripherals/dshot.md) is a digital ESC protocol that is highly recommended for vehicles that can benefit from reduce latency, in particular racing multicopters, VTOL vehicles, and so on.

Reviewer 2: ```suggestion
[DShot](../peripherals/dshot.md) is a digital ESC protocol that is highly recommended for vehicles that can benefit from reduced latency, in particular racing multicopters, VTOL vehicles, and so on.
```

It has reduced latency and is more robust than both [PWM](#pwm) and [OneShot](#oneshot-125).

> In addition it does not require ESC calibration, telemetry is available from some ESCs, and you can revers motor spin directions

Reviewer 2: ```suggestion
In addition it does not require ESC calibration, telemetry is available from some ESCs, and you can reverse motor spin directions.
```

PX4 configuration is done in the [Actuator Configuration](../config/actuators.md).

> Selecting a higher rate DShot ESC in the UI result in lower latency, but lower rates are more robust (and hence more suitable for large aircraft with longer leads); some ESCs only support lower rates (see datasheets for information).

Reviewer 2: ```suggestion
Selecting a higher rate DShot ESC in the UI results in lower latency, but lower rates are more robust (and hence more suitable for large aircraft with longer leads); some ESCs only support lower rates (see datasheets for information).
```

Setup:

- [ESC Wiring](../peripherals/pwm_escs_and_servo.md) (same as for PWM ESCs)
- [DShot](../peripherals/dshot.md) also contains information about how to send commands etc.

## DroneCAN

[DroneCAN ESCs](../dronecan/escs.md) are recommended when DroneCAN is the primary bus used for your vehicle.
The PX4 implementation is currently limited to update rates of 200Hz.

DroneCAN shares many similar benefits to [Dshot](#dshot) including high data rates, robust connection over long leads, telemetry feedback, no need for calibration of the ESC itself.

[DroneCAN ESCs](../dronecan/escs.md) are connected via the DroneCAN bus (setup and configuration are covered at that link).

## PWM

[PWM ESCs](../peripherals/pwm_escs_and_servo.md) are commonly used for fixed-wing vehicles and ground vehicles (vehicles that require a lower latency like multicopters typically use oneshot or dshot ESCs).

> PWM ESCs communicate using a periodic pulse, where the _width_ of the pulse indicates the desired power level.

Reviewer 2: ```suggestion
PWM ESCs communicate using a periodic pulse, where the _width_ of the pulse indicates the desired speed.
```

> The pulse wdith typically ranges between 1000uS for zero power and 2000uS for full power.

Reviewer 2: ```suggestion
The pulse width typically ranges between 1000uS for zero power and 2000uS for full power.
```
